File: CucumberJVM/Chapter14/14.7-ValidationTesterPattern/Before/src/main/java/com/wimp/app/models/Order.java
/*
 * NOTE: This application ("WIMP - Where Is My Pizza") is provided solely to
 * demonstrate the Behavior-Driven Development scenario automation patterns
 * described in the book "Effective Behavior-Driven Development" by
 * Gaspar Nagy and Seb Rose.
 *
 * It is NOT a complete or production-ready implementation. It deliberately
 * uses shortcuts and simplifications (e.g. authentication, data storage,
 * error handling, security) that are NOT suitable for a real application.
 * Do not use this code as a basis for production software.
 */

package com.wimp.app.models;

import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;

public class Order {
    private int orderNo;
    private String customerName;
    private String customerEmail;
    private final List<PizzaItem> items = new ArrayList<>();
    private OrderStatus status;
    private LocalDateTime placingTime;
    private String deliveryAddress;
    private DeliveryMethod deliveryMethod;
    private ContactDetails contactDetails;

    public int getOrderNo() {
        return orderNo;
    }

    public void setOrderNo(int orderNo) {
        this.orderNo = orderNo;
    }

    public String getCustomerName() {
        return customerName;
    }

    public void setCustomerName(String customerName) {
        this.customerName = customerName;
    }

    public String getCustomerEmail() {
        return customerEmail;
    }

    public void setCustomerEmail(String customerEmail) {
        this.customerEmail = customerEmail;
    }

    public List<PizzaItem> getItems() {
        return items;
    }

    public void addItem(PizzaItem pizzaItem) {
        items.add(pizzaItem);
    }

    public void clearItems() {
        items.clear();
    }

    public OrderStatus getStatus() {
        return status;
    }

    public void setStatus(OrderStatus status) {
        this.status = status;
    }

    public LocalDateTime getPlacingTime() {
        return placingTime;
    }

    public void setPlacingTime(LocalDateTime placingTime) {
        this.placingTime = placingTime;
    }

    @Override
    public String toString() {
        return "Order[orderNo=%d, customerName=%s, status=%s, items=%d]"
            .formatted(orderNo, customerName, status, items.size());
    }

    public String getDeliveryAddress() {
        return deliveryAddress;
    }

    public void setDeliveryAddress(String deliveryAddress) {
        this.deliveryAddress = deliveryAddress;
    }

    public DeliveryMethod getDeliveryMethod() {
        return deliveryMethod;
    }

    public void setDeliveryMethod(DeliveryMethod deliveryMethod) {
        this.deliveryMethod = deliveryMethod;
    }

    public ContactDetails getContactDetails() {
        return contactDetails;
    }

    public void setContactDetails(ContactDetails contactDetails) {
        this.contactDetails = contactDetails;
    }
}

File: Reqnroll/Chapter14/14.4-TerminologyPromotionPattern/After/WIMP.App/Models/Order.cs
/*
 * NOTE: This application ("WIMP - Where Is My Pizza") is provided solely to
 * demonstrate the Behavior-Driven Development scenario automation patterns
 * described in the book "Effective Behavior-Driven Development" by
 * Gaspar Nagy and Seb Rose.
 *
 * It is NOT a complete or production-ready implementation. It deliberately
 * uses shortcuts and simplifications (e.g. authentication, data storage,
 * error handling, security) that are NOT suitable for a real application.
 * Do not use this code as a basis for production software.
 */

namespace WIMP.App.Models;

public class Order
{
    public int OrderNo { get; set; }
    public OrderStatus Status { get; set; } = OrderStatus.New;
    public DateTimeOffset PlacingTime { get; set; }
    public string CustomerName { get; set; } = null!;
    public string? DeliveryAddress { get; set; }

    private readonly List<PizzaItem> items = new();
    public IReadOnlyList<PizzaItem> Items => items.AsReadOnly();

    public void AddItem(PizzaItem pizzaItem)
    {
        items.Add(pizzaItem);
    }

    public void ClearItems()
    {
        items.Clear();
    }
}
